src/js/components/ServiceDetail.js

```diff
@@ -22,8 +31,53 @@ class ServiceDetail extends mixin(InternalStorageMixin, TabsMixin) {
     };
 
     this.state = {
-      currentTab: Object.keys(this.tabs_tabs).shift()
+      currentTab: Object.keys(this.tabs_tabs).shift(),
+      progressDetailModalOpen: false,
+      servicePlan: null
     };
+
+    this.store_listeners = [{
+      name: 'servicePlan',
+      events: [
+        'error'
+      ]
+    }];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    ServicePlanStore.addPlanChangeListener(this.props.service.id,
+      this.onServicePlanStoreChange);
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount(...arguments);
+    ServicePlanStore.removePlanChangeListener(this.props.service.id,
+      this.onServicePlanStoreChange);
+  }
+
+  onServicePlanStoreChange() {
+    this.setState({
+      servicePlan: ServicePlanStore.getServicePlan(this.props.service.id)
+    });
+  }
+
+  onServicePlanStoreError() {
+    this.setState({
+      servicePlan: null
```

Comment:
Disclaimer: I don't see this in the previous PRs, so I'm going to comment here.
This could be in one line too